plans/phase-1/multigrescluster-cr-definition.md

```diff
@@ -0,0 +1,713 @@
+                    limits:
+                      cpu: "400m"
+                      memory: "256Mi"
+            - cell: "us-west-2"
+              replica:
+                replicas: 3
+                dataVolumeClaimTemplate:
+                  accessModes: ["ReadWriteOnce"]
+                  resources:
+                    requests:
+                      storage: "100Gi"
+                postgres:
+                  resources:
+                    requests:
+                      cpu: "1"
+                      memory: "2Gi"
+                    limits:
+                      cpu: "2"
+                      memory: "4Gi"
+                multipooler:
+                  resources:
+                    requests:
+                      cpu: "200m"
+                      memory: "128Mi"
+                    limits:
+                      cpu: "400m"
+                      memory: "256Mi"
+# --- Status ---
+status:
+  observedGeneration: 1
+  globalTopoServer:
+    etcd:
+      available: "True"
+  conditions:
+    - type: Available
+      status: "True"
+      lastTransitionTime: "2025-10-08T12:00:00Z"
+      message: "All components are available."
+    - type: Progressing
+      status: "False"
+      lastTransitionTime: "2025-10-08T12:00:00Z"
+      message: "Cluster reconciliation complete."
+  cells:
+    us-east-1:
+      gatewayAvailable: "True"
+      topoServerAvailable: "True"
+    us-west-2:
+      gatewayAvailable: "True"
+  databases:
+    users_db:
+      desiredInstances: 6
+      readyInstances: 6
+      servingWrites: "True"
+  admin:
+    available: "True"
+    serviceName: "example-multigres-cluster-admin"
+  orchestrator:
+    available: "True"
+```
+
+### Key Design Decisions for Discussion
+
+This section outlines critical design questions where we seek collaboration with the Supabase team to ensure the final API meets both immediate and long-term requirements.
+
+#### 1\. Defining a Declarative Sharding Strategy
+
+A core feature of Multigres is its ability to scale horizontally through sharding via `TableGroups`. The operator must provide a powerful yet intuitive way for users to define this sharding strategy declaratively. The central question is how to map the logical concepts of `TableGroups` and shards to the physical deployment of pods across different cells.
+
+We propose a model which separates the logical partitioning strategy from the physical pod configuration. This involves defining `tableGroups` within the database spec, each with a `partitioning` strategy and a `shardTemplate`.
+
+Below is our proposed API structure for this section. It establishes a clear pattern for defining sharded databases and serves as a foundation for more advanced partitioning strategies in the future.
+
+```yaml
+# Proposal for a shards snippet for a MultigresCluster CRD
+spec:
+  databases:
+    managed: true
+    templates:
+      - name: "production_db"
+        spec:
+          tableGroups:
```

Comment:
what happens when both `spec.databases.templates[0].spec.tableGroups` and `spec.databases.templates[0].spec.pools` are defined?

perhaps it's better to only accept the tableGroups structure with sensible defaults, so that it's easy to configure a non-sharding cluster.